packages/shared/src/clients/rpc/RpcClient2.test.ts

```diff
@@ -48,6 +55,100 @@ describe(RpcClient2.name, () => {
     })
   })
 
+  describe(RpcClient2.prototype.call.name, () => {
+    it('calls eth_call with correct parameters', async () => {
+      const http = mockObject<HttpClient2>({
+        fetch: async () => ({
+          result: '0x123abc',
+        }),
+      })
+      const rpc = mockClient({ http })
+
+      const result = await rpc.call(
+        {
+          to: EthereumAddress('0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'),
+          data: Bytes.fromHex('0x70a08231'),
+        },
+        'latest',
+      )
+
+      expect(result).toEqual(Bytes.fromHex('0x123abc'))
+      expect(http.fetch).toHaveBeenCalledTimes(1)
+      //@ts-expect-error
+      expect(http.fetch.calls[0].args[1]?.body).toMatchRegex(
```

Comment:
you can pass custom generateId() function, and test whole body without this regex